lib/multi-level.js

```diff
@@ -120,13 +119,23 @@ class MultiLevelCache {
           expiry: Date.now() + this._maxAge
         };
 
-        // Out-of-band cache update
-        // NB: an in-band update would `await` this Promise.all block
-        Promise.all(this._caches.map(cache => cache.set(key, cacheItem))).catch(err => {
-          throw new Error(`Error caching ${key}`, err);
-        }).then(() => {
+        const willCache = shouldCache || this.shouldCache;
+        if (typeof willCache !== 'function') {
```

Comment:
IMHO do this in the ctor and then do it again in the get call and make it something like 
```js
if (shouldCache && typeof shouldCache !== 'function') {
... throw
}
```

Then here you just do the falsey-coalesce and just call the function.  That'll save on doing a type check everytime unless a get fn was actually set and will throw instantly with a good callstack rather than after the update has happened. (i.e. make it easier for the dev to catch the error sooner and easier for them to understand)